src/libs/ReportUtils.ts

```diff
@@ -3527,6 +3527,13 @@ function getInvoicesChatName(report: OnyxEntry<Report>, receiverPolicy: OnyxEntr
     return getPolicyName(report, false, invoiceReceiverPolicy);
 }
 
+const reportNameCache = new Map<string, {lastVisibleActionCreated: string; reportName: string}>();
+
+/**
+ * Get a cache key for the report name.
+ */
+const getCacheKey = (report: OnyxEntry<Report>): string => `${report?.reportID}-${report?.lastVisibleActionCreated}-${report?.reportName}`;
+
```

Comment:
NAB: If the cache key contains all the information stored in the value, can we use a set instead of a map? Would it increase performance enough to be worth it?

Reply:
Using a `Set` instead of a `Map` wouldn’t significantly improve performance in this case. Also a `Map` is more appropriate because it allows for quickly retrieve values based on keys, which is necessary for caching. While a `Set` ensures uniqueness, it doesn’t provide the key-value pairing needed for efficient caching here.